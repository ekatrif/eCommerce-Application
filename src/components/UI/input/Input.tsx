import React from 'react';
import classes from './Input.module.css';

interface InputProps {
  value?: string;
  name: string;
  type: 'text' | 'password';
  placeholder?: string;
  onBlur?: (event: React.FocusEvent<HTMLInputElement>) => void;
  onChange?: (event: React.ChangeEvent<HTMLInputElement>) => void;
}

const Input: React.FC<InputProps> = ({ onBlur, onChange, name, type, placeholder }) => {
  return (
    <input
      onBlur={onBlur}
      onChange={onChange}
      name={name}
      type={type}
      placeholder={placeholder}
      className={classes.input}
    />
  );
};

export default Input;
