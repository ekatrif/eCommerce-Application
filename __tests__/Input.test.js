import React from 'react';
import renderer from 'react-test-renderer';
import Input from '../src/components/UI/input/Input';

test('Input renders correctly', () => {
  const tree = renderer
    .create(<Input
      value="test-value"
      name="email"
      type="text"
      placeholder="Enter your email"
    />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
