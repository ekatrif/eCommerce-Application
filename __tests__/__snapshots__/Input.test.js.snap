// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders correctly 1`] = `
<input
  className="input"
  name="email"
  placeholder="Enter your email"
  type="text"
/>
`;
